### ISSUE TITLE
Protobuf 3.6

### ISSUE THREAD

**Issue by schelmo** [OpenLightingProject/ola#1463]
i dont know if its a problem with protobuf or ola:
```
checking for libprotobuf... yes
checking protobuf library version... 3.6.0
checking for protoc... /usr/bin/protoc
checking protoc version... 3.6.0
checking google/protobuf/compiler/command_line_interface.h usability... no
checking google/protobuf/compiler/command_line_interface.h presence... yes
configure: WARNING: google/protobuf/compiler/command_line_interface.h: present but cannot be compiled
configure: WARNING: google/protobuf/compiler/command_line_interface.h: check for missing prerequisite headers?
configure: WARNING: google/protobuf/compiler/command_line_interface.h: see the Autoconf documentation
configure: WARNING: google/protobuf/compiler/command_line_interface.h: section "Present But Cannot Be Compiled"
configure: WARNING: google/protobuf/compiler/command_line_interface.h: proceeding with the compiler's result
configure: WARNING: ## --------------------------------------------- ##
configure: WARNING: ## Report this to open-lighting@googlegroups.com ##
configure: WARNING: ## --------------------------------------------- ##
checking for google/protobuf/compiler/command_line_interface.h... no
configure: error: Cannot find the protoc header files 
```

**Comment by peternewman** [comment]
Quite possibly OLA. Can we see your config.log please @schelmo .

Also what OS and how did you install protobuf?

What version of OLA are you compiling against?

**Comment by schelmo** [comment]
currently im building against latest master ( https://aur.archlinux.org/packages/ola-git/ , PKGBUILD here: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ola-git )
protobuf version is: 3.6.0.1 ( https://www.archlinux.org/packages/extra/x86_64/protobuf/ )

[config.log](https://github.com/OpenLightingProject/ola/files/2280439/config.log)

**Comment by schelmo** [comment]
[config.log](https://github.com/OpenLightingProject/ola/files/2280443/config.log)

building against master on archlinux ( https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ola-git )
Latest protobuf version on archlinux is: 3.6.0.1 ( https://www.archlinux.org/packages/extra/x86_64/protobuf/ )

**Comment by peternewman** [comment]
How's your autotools @schelmo ? It looks from config.log like Protobuf 3.6 needs C++11 support.

You should be able to work around it for now by adding the environment variables listed in here:
https://github.com/OpenLightingProject/ola/issues/1244#issue-225410940

Ultimately we want to work around it by extending this:
https://github.com/OpenLightingProject/ola/blob/master/configure.ac#L67-L88

If you fancy hacking that around so it also checks for Protobuf >= 3.6, otherwise I'll try and come up with a branch for testing shortly.

**Comment by schelmo** [comment]
thx,
```with ac_cv_gnu_plus_plus_98="no"``` the ./configure runs just fine

**Comment by peternewman** [comment]
So the good news is Homebrew has protobuf 3.6, so we can fix and test this on Travis:
https://formulae.brew.sh/formula/protobuf

Ideally the configure changes want to be a bit smarter. If you use your workaround for now and I'll do a branch might be easier.

Does make and make check complete after that, and have you done some basic functional sanity tests to check we actually work with protobuf 3.6?

**Comment by peternewman** [comment]
A few notes for that PKGBUILD @schelmo . You could add support for --enable-rdm-tests if you install python-numpy too. Also we're on version 0.10.7 now (although master is a long way ahead of that).

**Comment by peternewman** [comment]
I was rather confused how this was working already, but we obviously run the unit tests on our Travis builds (which already run against Protobuf 3.6), but the presence of cppunit 1.14 also forces us into C++11 mode, hence why we hadn't noticed the issue.

Feel free to take a look at Travis, or better yet can you test https://github.com/OpenLightingProject/ola/pull/1467 please @schelmo ?

**Comment by peternewman** [comment]
Closed via #1467
